projects/bird-atlas/src/app/+grid/grid-info/grid-info.component.ts

```diff
@@ -1,9 +1,156 @@
+
+function getGeoJSONFeature(ykj: string) {
+  const langLngStr = ykj.split(':');
+  return convertYkjToGeoJsonFeature(langLngStr[0], langLngStr[1]);
+}
 
 @Component({
   selector: 'ba-grid-info',
-  template: ``
+  templateUrl: './grid-info.component.html',
+  styleUrls: ['./grid-info.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class GridInfoComponent {
+export class GridInfoComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('lajiMap', { static: false }) lajiMapElem: ElementRef;
+
+  private unsubscribe$ = new Subject<void>();
+  private loadMap$ = new BehaviorSubject<string>(undefined);
+
+  data$ = this.route.paramMap.pipe(
+    tap(() => {
+      this.breadcrumbs.setBreadcrumbName(BreadcrumbId.GridInfo, undefined);
+      this.loading = true;
+      this.cdr.markForCheck();
+    }),
+    switchMap(params => this.atlasApi.getGridElement(params.get('id'))),
+    catchError(() => of(undefined)),
+    map(elem => elem ? {
+      elem,
+      rows: elem.data.map((d, idx) => ({
+        idx: idx + 1,
+        species: d.speciesName,
+        code: (<string>d.atlasCode.key).match(/[0-9]+/)?.[0] || null,
+        class: d.atlasClass.value
+      })),
+      status: 200
+    } : {
+      elem: undefined,
+      rows: [],
+      status: 404
+    }),
+    tap(d => {
+      if (!d.elem) { return; }
+      this.breadcrumbs.setBreadcrumbName(
+        BreadcrumbId.GridInfo,
+        d.elem.coordinates
+      );
+      this.headerService.setHeaders({
+        title: `${d.elem.name} ${d.elem.coordinates} | ${this.translate.instant('ba.header.title')}`
+      });
+      this.loadMap$.next(d.elem.coordinates);
+      this.loading = false;
+    }),
+  );
+
+  cols: TableColumn[] = [
+    {
+      prop: 'idx',
+      name: '#',
+      resizeable: false,
+      sortable: true,
+      width: 75
+    },
+    {
+      prop: 'species',
+      name: 'Laji',
+      resizeable: false,
+      sortable: true,
+      width: 125
+    },
+    {
+      prop: 'code',
+      name: 'Indeksi',
+      resizeable: false,
+      sortable: true,
+      width: 75
+    },
+    {
+      prop: 'class',
+      name: 'Luokka',
+      resizeable: false,
+      sortable: true,
+      width: 225
+    }
+  ];
+
+  loading = true;
+  datatableClasses = datatableClasses;
+  map: any;
+
+  constructor(
+    private zone: NgZone,
+    private atlasApi: AtlasApiService,
+    private route: ActivatedRoute,
+    private breadcrumbs: BreadcrumbService,
+    private translate: TranslateService,
+    private headerService: HeaderService,
+    private cdr: ChangeDetectorRef
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.loadMap$.pipe(
+      takeUntil(this.unsubscribe$),
+      tap(() => { if (this.map) { this.map.destroy(); } }),
+      filter(d => d !== undefined)
+    ).subscribe(ykj => {
+      setTimeout(() => { // yield control to renderer before accessing lajiMapElem
+        this.zone.runOutsideAngular(() => {
+          this.map = new LajiMap({
```

Comment:
Why the map is created every time when the data changes and why the laji map component is not used?
`<laji-map [data]="data" [options]="mapOptions" [visible]="visible"></laji-map>`

Reply:
`LajiMapComponent` requires a lot of stuff that is not included in `LajiMapModule` and I didn't spend time refactoring it all...
eg.
* environment props has to include apiBase, loginUrl, systemID (note that bird-atlas has no login feature)
* bunch of other userservice dependencies
* provider for logger service

Reply:
Ok, that makes sense